tests/functional/instance_manager/test_instances.py

```diff
@@ -0,0 +1,189 @@
+#
+# -----------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -----------------------------------------------------------------------------
+
+import os
+import subprocess
+import time
+import unittest
+
+from codechecker_lib import instance_manager
+
+from libtest import env
+from . import set_event
+from . import start_server
+
+"""
+Instance manager tests.
+"""
+
+
+def run_cmd(cmd):
+    print(cmd)
+    proc = subprocess.Popen(cmd,
+                            stdout=subprocess.PIPE)
+
+    out, _ = proc.communicate()
+    print(out)
+    return proc.returncode
+
+
+class Instances(unittest.TestCase):
+    """
+    Server instance manager tests.
+    """
+
+    def setUp(self):
+        # Get the test workspace used to tests.
+        self._test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self._test_workspace)
+
+    def testInitialFile(self):
+        """Tests that initially az instance file is created."""
+
+        self.assertEqual(instance_manager.list(), [],
+                         "The initial instance file was not empty initially.")
+
+    def testServerStart(self):
+        """Started server writes itself to instance list"""
+
+        test_cfg = env.import_test_cfg(self._test_workspace)
+        codechecker_1 = test_cfg['codechecker_1']
+        start_server(codechecker_1, test_cfg, 1)
+
+        self.assertEqual(len(instance_manager.list()), 1,
+                         "The started server was not appended to the "
+                         "instance list.")
+
+        self.assertEqual(instance_manager.list()[0]['port'],
+                         codechecker_1['viewer_port'],
+                         "The written port in the instance list is not the "
+                         "one the server was started with!")
+
+        self.assertEqual(instance_manager.list()[0]['workspace'],
+                         self._test_workspace,
+                         "Server was brought up with invalid workspace.")
+
+    def testServerStartSecondary(self):
+        """Another started server appends itself to instance list"""
+
+        test_cfg = env.import_test_cfg(self._test_workspace)
+        codechecker_1 = test_cfg['codechecker_1']
+        codechecker_2 = test_cfg['codechecker_2']
+        start_server(codechecker_2, test_cfg, 2)
+
+        self.assertEqual(len(instance_manager.list()), 2,
+                         "The started server was not appended to the "
+                         "instance list.")
+
+        # Workspaces must match, servers were started in the same workspace.
+        instance_wses = [i['workspace'] for i in instance_manager.list()]
```

Comment:
This is a bit weird abbreviation for me `wses` for workspaces but I think I can live with this.